Sources/GDBRemote/DebugSessionImpl.cpp

```diff
@@ -1062,10 +1062,20 @@ ErrorCode DebugSessionImpl::onRemoveBreakpoint(Session &session,
   //    if (session.mode() != kCompatibilityModeLLDB)
   //        return kErrorUnsupported;
 
-  if (type != kSoftwareBreakpoint)
-    return kErrorUnsupported;
+  BreakpointManager *bpm = nullptr;
+  switch (type) {
+  case kSoftwareBreakpoint:
+    bpm = _process->softwareBreakpointManager();
+    break;
+
+  case kHardwareBreakpoint:
+  case kReadWatchpoint:
+  case kWriteWatchpoint:
+  case kAccessWatchpoint:
+    bpm = _process->hardwareBreakpointManager();
+    break;
+  }
```

Comment:
I'm going to commit this patch but I'll just add `case kBreakpointTypeMax` at the end to avoid incomplete switch warnings.